src/main.js

```diff
@@ -63,29 +79,37 @@ function conceal(htmlElement){
   htmlElement.classList.add("hidden")
 }
 
-
-function showMakeOwnPage() {
-
-  homePage.classList.add("hidden")
-  randomBtn.classList.add("hidden")
-  saveBtn.classList.add("hidden")
-
-  homeBtn.classList.remove("hidden")
-  makeOwnPage.classList.remove("hidden")
-
-}
-
-function viewSaved() {
-
-  homePage.classList.add("hidden")
-  randomBtn.classList.add("hidden")
-  saveBtn.classList.add("hidden")
-
-  homeBtn.classList.remove("hidden")
-  viewSavedPage.classList.remove("hidden")
-
+// "switchPage" hides the elements that should be hidden for a given page,
+// and reveals the items that should be revealed for that page
+// the resulting experience for the user is as if they had navigated to that page
+function switchPage(page){
+  var longestLength;
+  // make sure we run the loop at least enough times to visit every element in the longer of the two arrays (which will also be enough for the shorter of the two)
+  page.hidden.length > page.visible.length ? (longestLength = page.hidden.length) : (longestLength = page.visible.length)
+
+  // if there's an element in either array at any index between 0 and the longest index between the two,
+  // if it's in the hidden array, hide it. if it's in the visible array, show it.
+  for(var i = 0;i < longestLength;i++){
+    page.hidden[i] ? conceal(page.hidden[i]) : console.log('No item at hidden[i]')
+    page.visible[i] ? reveal(page.visible[i]) : console.log('condition 2')
+  }
```

Comment:
@NickLiggett @GOECHA we should review how this works over zoom cuz it's a little confusing